fix: sets `buildPath` to `dist`

moltar opened the pull request:
Closes: STA-20

---

### All Submissions:

- [x] Have you followed the guidelines in our [Contributing guide?](https://github.com/StackBakery/stackgen/blob/master/CONTRIBUTING.md)

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license_

moltar committed:
fix: sets `buildPath` to `dist`

moltar committed:
fix: use workspace dependency resolution

moltar committed:
docs: use jsdoc instead of comments

Pagebakers commented on .stackgenrc.ts:
```diff
@@ -135,7 +137,7 @@ Bindings
     new ManifestEntry(project, 'CompileScript', {
       scripts: {
-        compile: 'tsup ./index.ts',
+        compile: `tsup --out-dir ${buildPath} ./index.ts`,
```
tsup picks this up from tsconfig right?

moltar replied:
The `--out-dir` value? 

Not sure. It's not documented in the docs, only as `--help` flag:

```
  -d, --out-dir <dir>          Output directory (default: dist)
```

So by sound of that, I think it's not using tsconfig values.

moltar committed:
fix: cli compilation (#2)

* chore: installs tsx

* refactor: use `tsx` instead of `ts-node`

* chore: disable `wrapCommand` for now as it breaks

TypeError: Cannot set property handler of #<Object> which has only a getter

* chore: uninstalls ts-node

* chore: mv `typescript` to `devDependencies`

It is no longer needed at runtime as we use `tsx`

moltar committed:
Merge branch 'main' into roman/sta-20-set-dist-dir-of-packages-to-dist
